### PR TITLE
[EC-416] Refactor organization permission checks

### PR DESCRIPTION
**Pull request by eliykat**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Our client-side organization permission checks are difficult to understand and maintain.  This PR refactors them to:
* combine and streamline permission checks
* remove duplicate ways of checking permissions
* simplify logic

It also removes the SCIM feature flag because it was in the way, and it needed to be done anyway (EC-363).

## Code changes

- **apps/web/src/app/organizations/guards/permissions.guard.ts**
  - instead of expecting an array of `Permission` enum values, this guard now expects a callback which returns a boolean, indicating whether the user has the permissions required. This is more flexible and means we don't need the `Permissions` enum.
  - if the user doesn't have the required permissions, return them to the org vault if possible. Just a small quality of life improvement.
- **bitwarden_license/bit-web/src/app/providers/guards/permissions.guard.ts** - rename and make equivalent changes
- **libs/common/src/enums/permissions.ts** - delete, no longer required
- **libs/common/src/models/domain/organization.ts**
  - remove `hasAnyPermission`, which is no longer required
  - when checking `canManage<Feature>`, also check that the organization has the feature enabled. You cannot manage a feature that your organization cannot use. This avoids having to check both (`canManageSso && useSso` etc) which is easy to forget (see #3250). I couldn't see anywhere that combining this logic would be a problem.
- **apps/web/src/app/organizations/navigation-permissions.ts** - this replaces the old `NavigationPermissionsService`, which was not really a service, and doesn't even need to be a class. It just houses some permission checks that are shared between a few different components/services.
- **routing modules** (various) - replace `Permissions` enums with equivalent callbacks
- **manage components** (various) - remove permission checks in `ngOnInit`, these belong in guards and are covered by the guard logic
- **apps/web/src/app/organizations/manage/manage.component.ts** - remove `canUse...` checks that are now incorporated into the `canManage...` logic.

## Before you submit

```
- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

**Review by Hinton — COMMENTED** [review]
Great work, I think this simplifies things a fair bit.

1. Should we bite the bullet and rename the permission guards to identify what they guard against? It's confusing to have multiple `PermissionsGuards`.
2. It's also quite confusing to have use the same data object with different guards since they use a lambda with different data models.

On another note, this would have been much nicer if Angular just supported lambdas directly in `canActivate` instead of routing everything through their DI.

File: bitwarden_license/bit-web/src/app/organizations/organizations-routing.module.ts [hunk 1]
```diff
@@ -23,23 +23,23 @@ const routes: Routes = [
...
+          permissions: canAccessManageTab,
         },
         children: [
           {
             path: "sso",
             component: SsoComponent,
             canActivate: [PermissionsGuard],
             data: {
-              permissions: [Permissions.ManageSso],
+              permissions: (org: Organization) => org.canManageSso,
```
[the remark this answers is not in the document]
Comment (eliykat) [reply bitwarden_license/bit-web/src/app/organizations/organizations-routing.module.ts]: This is done

**Comment by eliykat** [comment]
Additional changes:
* rename the permissions guards to `org-permissions.guard.ts` and `provider-permissions.guard.ts` with similar changes to the class names
* fold `provider.guard.ts` into `provider-permissions.guard.ts` as they are quite similar, we already did this for the org guards
* add tests

**Review by shane-melton — COMMENTED** [review]
This looks good to me and should also make the permissions in the org admin refresh more straightforward as well! 

Only thing I see is storybook seems to be failing on this branch and causing the chromatic check to fail.

**Comment by eliykat** [comment]
Maybe just flaky CI - let me mash the "re-run" button a few times and see if that fixes it :D

**Comment by eliykat** [comment]
@shane-melton The build is fixed, I had to pin `jest-mock-extended` at `v2.0.6` due to a bug: https://github.com/marchaos/jest-mock-extended/issues/97

**Review by shane-melton — APPROVED** [review]

**Pull request merged**